dali/python/nvidia/dali/plugin/base_iterator.py

```diff
@@ -252,8 +252,18 @@ def _get_outputs(self):
             if self._size < 0 and self._auto_reset:
                 self.reset()
             raise e
+        for out in outputs:
+            self._check_batch_size(out)
         return outputs
 
+    def _check_batch_size(self, out):
+        if self._reader_name or self._size != -1:
```

Comment:
Minor optimization: move this check outside the loop over outputs.

Reply:
Done